torch/_inductor/fx_passes/split_cat.py

```diff
@@ -1002,3 +1040,287 @@ def simplify_split_cat(match: Match, split_sections: List[int], dim: int):
         return
     split_node = next(node for node in match.nodes if node.target == torch.split)
     SplitCatSimplifier().simplify(match.graph, split_node, split_sections)
+
+
+'''
+############The pattern to be optimized is#########
+
+                split_node(dim=1)
+      /     \         ...       /         \
+getitem    getitem          getitem     getitem   -> user=1
+   \       /                     \       /
+     cat (user=mul, dim=1)           cat(user=mul, dim=1)
+      |            \                   |          \
+
+################After transformation#############
+
+                split_node(dim=1)
+      /              ...                  \
+    getitem                             getitem
+    |    \                              |     \
+'''
+def has_single_user(node: torch.fx.Node):
+    return len(node.users) == 1
+
+
+def safe_to_abort_node(node: torch.fx.Node):
+    """
+    1. the input nodes of the node should come from the same parent
+    2. the user of all the input nodes should be only one
+    """
+    prev_node = None
+    for arg in node.args[0]:
+        if not has_single_user(arg):
+            return False
+        if prev_node is None:
+            prev_node = arg.args[0]
+        else:
+            if arg.args[0] != prev_node:
+                return False
+    return True
+
+
+def remove_zeros(split_sections: List[int]):
+    """
+    Remove zeros from the list and get the index mapping dict from getitem
+    in split node to getitem in new split node
+    """
+    new_split_sections, index_mapping = [], {}
+    idx = 0
+    for i in range(len(split_sections)):
+        if split_sections[i] > 0:
+            new_split_sections.append(split_sections[i])
+            index_mapping[i] = idx
+            idx += 1
+
+    return new_split_sections, index_mapping
+
+
+@register_graph_pattern(
+    CallFunction(
+        torch.cat,
+        getitem_split,
+        dim=Ignored(),
+        _users=MULTIPLE,
+    ),
+    pass_dict=merge_getitem_cat_pass,
+    extra_check=config_flag("split_cat_fx_passes"),
+)
+def merge_getitem_cat(match: Match, split_sections: List[int], dim: int):
```

Comment:
Is this function duplicated with #110923? Probably diff exporting messed up the change stack.

Reply:
Yes. Will rebase and rerun the diff.